zcash_test_vectors/zip_0244.py

```diff
@@ -177,23 +177,22 @@ def orchard_actions_compact_digest(tx):
         digest.update(bytes(desc.nullifier))
         digest.update(bytes(desc.cmx))
         digest.update(bytes(desc.ephemeralKey))
-        digest.update(desc.encCiphertext[:ciphertext_offset(tx.nVersionGroupId)['compact_end']])
+        digest.update(desc.encCiphertext[:ciphertext_offset(tx.version_bytes())['compact_end']])
     return digest.digest()
 
 def orchard_actions_memos_digest(tx):
     digest = blake2b(digest_size=32, person=b'ZTxIdOrcActMHash')
-    compact_end = ciphertext_offset(tx.nVersionGroupId)['compact_end']
-    memo_end = ciphertext_offset(tx.nVersionGroupId)['memo_end']
+    offsets = ciphertext_offset(tx.version_bytes())
     for desc in tx.vActionsOrchard:
-        digest.update(desc.encCiphertext[compact_end:memo_end])
+        digest.update(desc.encCiphertext[offsets['compact_end']:offsets['memo_end']])
```

Comment:
call it `o` and use dot notation to make it more compact.
`o.compact_end:o.memo_end`

Reply:
Dot notation is not present for dictionaries, defined an Offsets class instead, then did the above.

Reply:
right, we can use `namedtuple`
```python
from collections import namedtuple                                                                                                                                                                                                            
                                                                                                                                                                                                                                               
 def get_data():                                                                                                                                                                                                                               
     Person = namedtuple('Person', ['name', 'age'])                                                                                                                                                                                            
     return Person(name='Alice', age=30)                                                                                                                                                                                                       
                                                                                                                                                                                                                                               
 data = get_data()                                                                                                                                                                                                                             
 print(data.name)  # Output: Alice                                                                                                                                                                                                             
 print(data.age)   # Output: 30   
```

or go back to returning a tupple but make the variables named:
```python
def get_data():                                                                                                                                                                                                                                
    name = 'Alice'                                                                                                                                                                                                                             
    age = 30                                                                                                                                                                                                                                   
    return name, age
```

Reply:
Used `namedtuple`, that looks better.